### ISSUE TITLE
Navigation Sidebar: If there are unsaved changes, show the save modal when clicking on  "Manage all pages"

### ISSUE THREAD

**Issue by carolinan** [WordPress/gutenberg#50648]
## What problem does this address?

When you go into the Site Editor, Select "Pages" and then click on "Manage all pages",  the browser default "leave site" popup shows, and you are taken to the page screen in the WordPress admin area.

<img width="1505" alt="Leave site popup" src="https://github.com/WordPress/gutenberg/assets/7422055/e830fbf1-9a5a-47bf-adea-d759685b0f93">

## What is your proposed solution?

Activating this "Manage all pages" link should check if there are unsaved changes, and if there are, display the save modal.
The save modal could have a "continue" button that takes you to the page screen.

**Comment by annezazu** [comment]
@WordPress/gutenberg-design curious for thoughts here.

**Comment by jasmussen** [comment]
Ah, you mean show the save modal _instead_ of showing this `confirm` dialog? That could work, but we'd likely need some message title in the modal showing why it suddenly pops up, and a clearer way to intentionally _discard_ changes.

Right now discarding changes is as simple as pressing "Leave". If we move towards showing the save dialog, we have to have an equally easy way to do that.

**Comment by carolinan** [comment]
Yes, but only when there actually are unsaved changes. The current modal does not give me the option to save, only leave or close.

**Comment by jameskoster** [comment]
This behavior is consistent with all other links that exit the editor environment. That's not to say we shouldn't change it, just that we should consider all such events before making a specific solution for this one scenario.

One potential drawback to the proposed solution is that there would be no way to exit the editor and discard all of your changes, except for closing the browser tab which seems a little extreme. 

And speaking of closing the browser tab, a small benefit of the current browser-based dialog is that it will alert you of unsaved changes before closing the tab / browser which can be a life-saver. I don't know if it's possible for in-app alerts to behave this way?

My initial thought is that there should be an interstitial alert that appears if there are unsaved changes when you exit the editor, or close the tab. It would ask what you want to do:

* Leave without saving → Head to destination / close tab
* Review changes → Open save modal
* Cancel → Stay where you are

**Comment by carolinan** [comment]
I am closing this since the page screens and the sidebar has changed and this is not an issue with the flow described here.
